### ISSUE TITLE
Compile custom julia sysimage

### ISSUE THREAD

**Issue by davidanthoff** [jupyter/repo2docker#601]
Once https://github.com/jupyter/repo2docker/pull/595 is merged and we have some experience with it, we should investigate whether we can use [PackageCompiler.jl](https://github.com/JuliaLang/PackageCompiler.jl) during the build step to compile a custom sysimage that includes all the packages in the `Project.toml`. That custom sysimage would then be stored in the docker file. This has the potentially to make things _a lot_ more snappy because we could probably get around a fair bit of JIT compiling.

**Comment by NHDaly** [comment]
woahhhhh that'd be super duper cool! :D I love it.

**Comment by asinghvi17** [comment]
Beware of issues that can arise due to this, though - Makie, for example, will not show a plot display unless `AbstractPlotting.__init__()` is called first (because of weirdness around the display stack).  There might be other examples of this behaviour that you'd have to watch out for.

**Comment by davidanthoff** [comment]
Yes, we should probably wait until this whole stack is more mature on the julia side of things. I think we can really only consider using it when PackageCompiler.jl is at a point where "it just works", not matter what.
